fieldpath/set.go

```diff
@@ -129,12 +129,41 @@ func (s *Set) RecursiveDifference(s2 *Set) *Set {
 // fields in s, as well as all the named fields that are typically not
 // included. For example, a set made of "a.b.c" will end-up also owning
 // "a" if it's a named fields but not "a.b" if it's a map.
+//
+// Missing keys in list maps are backfilled with null values.
```

Comment:
let's stop adding synthetic nulls everywhere and *just* focus on making sure callers / comparisons correctly handle non-identical key entries in sets